libtcmu.c

```diff
@@ -835,6 +835,48 @@ void tcmu_dev_set_private(struct tcmu_device *dev, void *private)
 	dev->hm_private = private;
 }
 
+const char *tcmu_dev_get_uio_name(struct tcmu_device *dev)
+{
+	return dev->dev_name;
+}
+
+void tcmu_set_thread_name(const char *prefix, struct tcmu_device *dev)
+{
+	const char *uio = dev ? tcmu_dev_get_uio_name(dev) : NULL;
+	char *pname;
+	int len;
+
+	if (!prefix) {
+		tcmu_dev_err(dev, "failed to set name for thread %lu\n",
+			     pthread_self());
+		return;
+	}
+
+	len = strlen(prefix);
+
+	if (uio)
+	       len += strlen(uio) + 1; // for '-'
+
+	if (len >= TCMU_THREAD_NAME_LEN) {
+		tcmu_dev_warn(dev, "the length for prefix(%s) + uio(%s) is (%d) should < (%d)\n",
+			      prefix, uio, len, TCMU_THREAD_NAME_LEN);
+		tcmu_dev_warn(dev, "will truncate the thread name to %d\n",
+			      TCMU_THREAD_NAME_LEN - 1);
+
+		len = TCMU_THREAD_NAME_LEN;
+	}
+
+	if (asprintf(&pname, "%s%s%s", prefix, uio ? "-" : "",  uio ? uio : "") == -1) {
+		tcmu_dev_err(dev, "no memory for pname\n");
+		return;
+	}
+
+	pname[len] = '\0';
```

Comment:
If len=TCMU_THREAD_NAME_LEN will this put the null at one char past the max length. The max length is 16 including the terminating null, so you would want to put it at pname[15] right?